api/grpcserver/admin_service.go

```diff
@@ -92,8 +94,9 @@ func (a AdminService) CheckpointStream(req *pb.CheckpointStreamRequest, stream p
 	}
 }
 
-func (a AdminService) Recover(_ context.Context, _ *pb.RecoverRequest) (*empty.Empty, error) {
-	a.logger.Panic("going to recover from checkpoint")
+func (a AdminService) Recover(ctx context.Context, _ *pb.RecoverRequest) (*empty.Empty, error) {
+	ctxzap.Info(ctx, "going to recover from checkpoint")
+	a.recover(ctx)
 	return &empty.Empty{}, nil
 }
```

Comment:
Should we remove this endpoint? Otherwise someone might call it and crash their node with it. We can re-add it when the functionality is available?

Or we could just make it a no-op.

Reply:
I thought this is used... Anyway, I only changed the way it kills the app to:
- avoid unnecessary stack trace (it's not an error after all),
- allow controlling what happens (mostly for testing).

> someone might call it and crash their node with it

This is exactly the purpose of this endpoint, no? :D Btw, it's in the _private_ endpoint group.

Reply:
I'm not sure, I'd like to hear @countvonzero's opinion on it:

- Should we leave the endpoint shutting down the node
- Update it to a no-op (and maybe just log that not yet available)
- Remove it completely

?

Reply:
Then OK, but I wouldn't pass in the `recover` function from the outside. You can set the field to

```golang
s.recover = func(context.Context) {
	// Allow time for the response to be sent.
	time.Sleep(time.Second)
	os.Exit(0)
}
```

in `NewAdminService` and overwrite it in the one test that calls it?

Reply:
One more thing: just exiting doesn't clean up `lock` files as far as I know? Does this maybe leave the node in a corrupted state or cause data loss if called at the wrong moment?

Reply:
I'm not sure. I think file locks are released when locked file descriptors are closed. But what happens if the application simply exits/crashes? Is the behavior different on a `panic()` vs `os.Exit()`?

Reply:
Both exit immediately as far as I know. I'm not sure if the library we are using for lock files registers some handler that removes the file when the process exits 🤔.

Reply:
Since no issue came up so far regarding this, it should be OK.